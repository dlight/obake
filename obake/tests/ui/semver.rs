#[obake::versioned]
#[obake(version(not_a_sem_str))]
struct Foo {}

#[obake::versioned]
#[obake(version("not a semver"))]
struct Bar {}

#[obake::versioned]
#[obake(version("0.1.0"))]
struct Baz {
    #[obake(cfg(not_a_ver_str))]
    field_0: u32,
}

#[obake::versioned]
#[obake(version("0.1.0"))]
struct Flim {
    #[obake(cfg("not a semver constraint"))]
    field_0: u32,
}

fn main() {}
