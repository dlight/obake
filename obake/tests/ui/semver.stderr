error: expected string literal
 --> $DIR/semver.rs:2:17
  |
2 | #[obake(version(not_a_sem_str))]
  |                 ^^^^^^^^^^^^^

error: unexpected character 'n' while parsing major version number
 --> $DIR/semver.rs:6:17
  |
6 | #[obake(version("not a semver"))]
  |                 ^^^^^^^^^^^^^^

error: expected string literal
  --> $DIR/semver.rs:12:17
   |
12 |     #[obake(cfg(not_a_ver_str))]
   |                 ^^^^^^^^^^^^^

error: unexpected character 'n' while parsing major version number
  --> $DIR/semver.rs:19:17
   |
19 |     #[obake(cfg("not a semver constraint"))]
   |                 ^^^^^^^^^^^^^^^^^^^^^^^^^
